src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php

```diff
@@ -821,6 +822,30 @@ public function testValidateUniquenessWithEmptyIterator($entity, $result)
         $this->assertNoViolation();
     }
 
+    /**
+     * @dataProvider invalidValuesProvider
+     */
+    public function testCannotValidateNonObject($value)
+    {
+        $constraint = new UniqueEntity([
+            'message' => 'myMessage',
+            'fields' => ['name'],
+            'em' => self::EM_NAME,
+        ]);
+
+        $this->expectException(UnexpectedValueException::class);
+
+        $this->validator->validate($value, $constraint);
+    }
+
+    public static function invalidValuesProvider(): array
+    {
+        return [
+            [null],
```

Comment:
`null` must always pass the validation. We explicitly combine a constraint with `NotNull` or `NotBlank` if we don't want `null` to be valid.

Reply:
Oops, fixed.